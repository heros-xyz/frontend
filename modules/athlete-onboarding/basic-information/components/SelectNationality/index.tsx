import { Box, Text } from "@chakra-ui/react";
import { useState } from "react";
import { useFormikContext } from "formik";
import { ArrowRight } from "@/components/svg/ArrowRight";
import Select from "@/components/common/Select";
import { NationalityIcon } from "@/components/svg/NationalityIcon";
import { useGetNationalityQuery } from "@/api/global";
import { filterSelectOptions } from "@/utils/functions";
import HerosOnboardingWrapperNew from "@/components/ui/HerosOnboardingWrapperNew";
import { useGetNationalities } from "@/libs/dtl/nationalities";
import { IValuesTypes } from "../../hooks";

interface SelectNationalityProps {
  onSubmit: () => void;
}

const SelectNationality: React.FC<SelectNationalityProps> = ({ onSubmit }) => {
  const [submitted, setSubmitted] = useState(false);
  const { values, errors, handleSubmit, setFieldValue } =
    useFormikContext<IValuesTypes>();

  const { nationalitiesMapped: nationalityList } = useGetNationalities();

  const submitForm = () => {
    setSubmitted(true);
    handleSubmit();

    if (!errors.nationality && values.nationality) {
      onSubmit();
    }
  };

  return (
    <HerosOnboardingWrapperNew
      Icon={
        <NationalityIcon
          w={{ base: "88px", xl: "140px" }}
          h={{ base: "75px", xl: "118px" }}
          color={{ base: "#FFFAE8", xl: "accent.1" }}
        />
      }
      textButton="Proceed"
      IconButton={<ArrowRight />}
      onSubmit={submitForm}
      title="Basic information"
      bgIconColor="accent.1"
    >
      <Box mb={{ base: 4, lg: 20 }} color="primary">
        <Box fontSize={{ lg: "xl" }} fontWeight="bold" mb={3}>
          Select Nationality
          <Text as="span" color="error.dark">
            {" "}
            *
          </Text>
        </Box>
        <Box>
          <Select
            placeHolder="Nationality"
            value={values?.nationality}
            optionCount={5}
            options={nationalityList}
            onChange={(value) => setFieldValue("nationality", value)}
            errorMessage={"This is a required field"}
            isInvalid={Boolean(submitted && errors.nationality)}
            filterSelectOptions={filterSelectOptions}
            isDarkTheme
          />
        </Box>
      </Box>
    </HerosOnboardingWrapperNew>
  );
};

export default SelectNationality;
