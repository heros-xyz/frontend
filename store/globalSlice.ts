import { PayloadAction, createSlice } from "@reduxjs/toolkit";
import { HYDRATE } from "next-redux-wrapper";

const initialState = {
  isLoading: false,
  isMute: true,
  isPlaying: false,
  accessToken: '',
  refreshToken: ''
};

export const appState = createSlice({
  name: "appState",
  initialState: initialState,
  reducers: {
    startLoading: (state) => {
      state.isLoading = true;
    },
    finishLoading: (state) => {
      state.isLoading = false;
    },
    playVideo: (state) => {
      state.isPlaying = true;
    },
    pauseVideo: (state) => {
      state.isPlaying = false;
    },
    setAccessToken: (state, action: PayloadAction<string>) => {
      state.accessToken = action.payload
    },
    setRefreshToken: (state, action: PayloadAction<string>) => {
      state.refreshToken = action.payload
    }
  },
  extraReducers: {
    [HYDRATE]: (state, action) => {
      if (action.payload.appState.accessToken) {
        state.accessToken = action.payload.appState.accessToken;
      }

      if (action.payload.appState.refreshToken) {
        state.refreshToken = action.payload.appState.refreshToken;
      }
    }
  }
});

export const { startLoading, finishLoading, setAccessToken, setRefreshToken } = appState.actions;

export default appState.reducer;
