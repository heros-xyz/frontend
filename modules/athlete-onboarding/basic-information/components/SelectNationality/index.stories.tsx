import type { Meta, StoryObj } from "@storybook/react";
import { Formik } from "formik";
import { Provider } from "react-redux";
import { store } from "@/store";
import SelectNationality from ".";

const meta: Meta<typeof SelectNationality> = {
  title: "Athlete/PageInfo/SelectNationality",
  component: SelectNationality,
  tags: ["autodocs"],
  parameters: {
    layout: "fullscreen",
  },
};

export default meta;
type Story = StoryObj<typeof SelectNationality>;

export const SelectNationalityStory: Story = {
  render: () => (
    <Provider store={store}>
      <Formik onSubmit={console.log} initialValues={{ nationality: "" }}>
        <SelectNationality onSubmit={console.log} />
      </Formik>
    </Provider>
  ),
};
