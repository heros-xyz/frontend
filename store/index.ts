import { configureStore } from "@reduxjs/toolkit";
import { createWrapper } from "next-redux-wrapper";
import { TypedUseSelectorHook, useDispatch, useSelector } from "react-redux";
import { setupListeners } from "@reduxjs/toolkit/query";
import globalSlice from "./globalSlice";

export const store = configureStore({
  reducer: {
    appState: globalSlice,

  },
  devTools: true,
  middleware: (gDM) => {
    return gDM().concat(

    );
  },
});

const makeStore = () => store;

export type AppStore = ReturnType<typeof makeStore>;
export type AppState = ReturnType<AppStore["getState"]>;
export type AppDispatch = AppStore["dispatch"];

export const wrapper = createWrapper<AppStore>(makeStore);
export const useAppDispatch = () => useDispatch<AppDispatch>();
export const useAppSelector: TypedUseSelectorHook<AppState> = useSelector;

setupListeners(store.dispatch);
