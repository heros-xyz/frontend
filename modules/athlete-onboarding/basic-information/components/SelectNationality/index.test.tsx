import { render } from "@testing-library/react";
import { Formik } from "formik";
import { Provider } from "react-redux";
import { store } from "@/store";
import SelectNationality from ".";

it("renders Select Nationality unchanged", () => {
  const { container } = render(
    <Provider store={store}>
      <Formik onSubmit={console.log} initialValues={{ nationality: "" }}>
        <SelectNationality onSubmit={console.log} />
      </Formik>
    </Provider>
  );
  expect(container).toMatchSnapshot();
});
