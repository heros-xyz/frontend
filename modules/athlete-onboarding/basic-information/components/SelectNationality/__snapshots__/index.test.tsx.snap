// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Select Nationality unchanged 1`] = `
<div>
  <div
    class="css-16tif5l"
  >
    <div
      class="chakra-container css-10klw3m"
    >
      <div
        class="css-12ij4x2"
      >
        BASIC INFORMATION
      </div>
      <div
        class="css-8qr4sb"
      >
        <div
          class="css-tayfq6"
        >
          <div
            class="css-6ywt87"
          >
            <svg
              class="chakra-icon css-h2mabo"
              focusable="false"
              viewBox="0 0 151 150"
              xmlns="http://www.w3.org/2000/svg"
            >
              <circle
                cx="75.5"
                cy="75"
                fill="white"
                r="74.5"
                stroke="#DDDDDD"
              />
              <g
                clip-path="url(#clip0_1913_25640)"
              >
                <path
                  d="M53.2849 44.6284C53.2849 44.6284 48.9421 48.2856 49.2849 51.257C49.5135 54.2284 51.4563 57.8856 53.5135 57.657C55.5706 57.4284 65.7421 52.3998 65.7421 52.3998L53.2849 44.6284Z"
                  fill="#FAC484"
                />
                <path
                  d="M53.2849 44.6284C53.2849 44.6284 48.9421 48.2856 49.2849 51.257C49.5135 54.2284 51.4563 57.8856 53.5135 57.657C55.5706 57.4284 65.7421 52.3998 65.7421 52.3998L53.2849 44.6284Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M62.1992 55.0285C62.1992 55.0285 56.942 57.6571 55.1134 57.8857C51.9134 58.4571 45.742 52.6285 51.7992 46.2285"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M53.8562 44.0569C53.8562 44.0569 66.7705 38.2284 69.9705 38.1141C73.2848 37.9998 87.2276 39.4855 90.1991 41.5427C93.1705 43.7141 118.542 63.9427 118.999 66.6855C119.456 69.3141 103.342 111.943 99.1133 112.286C94.8848 112.628 68.4848 101.657 65.8562 100.286C53.6276 93.4284 49.0562 87.8284 49.5133 87.1427C52.4848 81.8855 65.9705 66.9141 65.9705 66.9141C65.9705 66.9141 67.6848 55.2569 64.2562 52.2855C60.8276 49.3141 53.8562 44.0569 53.8562 44.0569Z"
                  fill="#FAC484"
                />
                <path
                  d="M79.3432 78.9141C79.3432 78.9141 82.7718 83.0283 80.8289 85.0855C78.886 87.1426 75.1146 84.6283 75.1146 84.6283C75.1146 84.6283 78.7718 88.6284 77.7432 91.0284C76.7146 93.4283 71.1146 91.9426 71.1146 91.9426C71.1146 91.9426 76.2575 96.6283 74.7718 99.1426C73.286 101.657 69.286 102.343 61.6289 96.7426"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M101.514 51.0283C101.514 51.0283 86.4281 29.9997 57.0566 43.2568"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M71.3438 54.114C74.2009 53.0854 76.6009 54.4568 76.6009 54.4568"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M72.9428 65.771C75.4571 54.571 68.4856 43.4853 57.2856 40.971C46.0856 38.3425 34.8856 45.4282 32.3713 56.6282C31.6856 59.5996 31.6856 62.6853 32.2571 65.5425L32.3713 65.8853C32.4856 66.3425 32.5999 66.6853 32.7142 67.1425L40.8285 98.3425C40.8285 98.4568 40.8285 98.4568 40.8285 98.571C41.1713 99.8282 42.1999 100.742 43.5713 101.085C44.9428 101.428 46.4285 100.857 47.2285 99.8282L68.0285 75.2568C68.3713 74.9139 68.7142 74.4568 69.0571 74.1139L69.1713 73.9996C70.8856 71.4853 72.2571 68.8568 72.9428 65.771ZM52.1428 63.8282C47.6856 62.7996 44.8285 58.3425 45.8571 53.771C46.8856 49.3139 51.3428 46.4568 55.9142 47.4853C60.3713 48.5139 63.2285 52.971 62.1999 57.5425C61.1713 62.1139 56.7142 64.8568 52.1428 63.8282Z"
                  fill="#F2696A"
                />
                <path
                  d="M72.9428 65.771C75.4571 54.571 68.4856 43.4853 57.2856 40.971C46.0856 38.3425 34.8856 45.4282 32.3713 56.6282C31.6856 59.5996 31.6856 62.6853 32.2571 65.5425L32.3713 65.8853C32.4856 66.3425 32.5999 66.6853 32.7142 67.1425L40.8285 98.3425C40.8285 98.4568 40.8285 98.4568 40.8285 98.571C41.1713 99.8282 42.1999 100.742 43.5713 101.085C44.9428 101.428 46.4285 100.857 47.2285 99.8282L68.0285 75.2568C68.3713 74.9139 68.7142 74.4568 69.0571 74.1139L69.1713 73.9996C70.8856 71.4853 72.2571 68.8568 72.9428 65.771ZM52.1428 63.8282C47.6856 62.7996 44.8285 58.3425 45.8571 53.771C46.8856 49.3139 51.3428 46.4568 55.9142 47.4853C60.3713 48.5139 63.2285 52.971 62.1999 57.5425C61.1713 62.1139 56.7142 64.8568 52.1428 63.8282Z"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M83.4573 62.4569C83.4573 62.4569 72.0287 58.4569 67.8001 61.314C63.5716 64.1712 63.5716 69.1998 65.5144 70.4569C67.4573 71.714 79.9144 75.1426 79.9144 75.1426"
                  fill="#FAC484"
                />
                <path
                  d="M83.4573 62.4569C83.4573 62.4569 72.0287 58.4569 67.8001 61.314C63.5716 64.1712 63.5716 69.1998 65.5144 70.4569C67.4573 71.714 79.9144 75.1426 79.9144 75.1426"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
                <path
                  d="M88.7147 52.3999C88.7147 52.3999 83.4576 54.7999 85.2862 62.457C85.2862 62.457 73.629 59.257 68.2576 60.9713C62.8862 62.6856 64.2576 67.9428 65.0576 69.5428C66.8862 73.3142 86.2005 76.7428 88.829 77.0856"
                  stroke="black"
                  stroke-miterlimit="10"
                  stroke-width="1.5"
                />
              </g>
              <defs>
                <clippath
                  id="clip0_1913_25640"
                >
                  <rect
                    fill="white"
                    height="74.2857"
                    transform="translate(31 38)"
                    width="88"
                  />
                </clippath>
              </defs>
            </svg>
          </div>
          <div
            class="css-yzqqny"
          >
            <div
              class="css-0"
            >
              <div
                class="css-1tilclj"
              >
                <div
                  class="css-johq1t"
                >
                  Select Nationality
                  <span
                    class="chakra-text css-1b8yx6g"
                  >
                     
                    *
                  </span>
                </div>
                <div
                  class="css-0"
                >
                  <div
                    class="css-s2ct51"
                  >
                    <div
                      class=" css-n9lnso-SelectContainer"
                    >
                      <span
                        class="css-1f43avz-a11yText-A11yText"
                        id="react-select-:r0:-live-region"
                      />
                      <span
                        aria-atomic="false"
                        aria-live="polite"
                        aria-relevant="additions text"
                        class="css-1f43avz-a11yText-A11yText"
                      />
                      <div
                        class=" css-6o1t9b-control"
                      >
                        <div
                          class=" css-15ki4p9-ValueContainer"
                        >
                          <div
                            class="css-xxdqwu"
                          >
                            <div
                              class="css-1yks822"
                            >
                              Nationality
                            </div>
                            <div
                              class=" css-119wo5m-Input"
                              data-value=""
                            >
                              <input
                                aria-autocomplete="list"
                                aria-describedby="react-select-:r0:-placeholder"
                                aria-expanded="false"
                                aria-haspopup="true"
                                autocapitalize="none"
                                autocomplete="off"
                                autocorrect="off"
                                class=""
                                id="react-select-:r0:-input"
                                role="combobox"
                                spellcheck="false"
                                style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                                tabindex="0"
                                type="text"
                                value=""
                              />
                            </div>
                          </div>
                        </div>
                        <div
                          class=" css-1hb7zxy-IndicatorsContainer"
                        >
                          <span
                            class=" css-1xjgjl1-IndicatorSeparator"
                          />
                          <svg
                            class="chakra-icon css-fgc0y"
                            focusable="false"
                            viewBox="0 0 20 20"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M16.25 6.875L10 13.125L3.75 6.875"
                              stroke="currentColor"
                              stroke-linecap="round"
                              stroke-linejoin="round"
                              stroke-width="1.5"
                            />
                          </svg>
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="css-lrby83"
              >
                <button
                  class="chakra-button css-k6pd2s"
                  type="button"
                >
                  Proceed
                  <span
                    class="css-9n3bbc"
                  >
                    <svg
                      class="chakra-icon css-iz364c"
                      focusable="false"
                      viewBox="0 0 15 14"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M1.25 7L13.75 7M13.75 7L8.125 1.375M13.75 7L8.125 12.625"
                        stroke="#33EFEF"
                        stroke-linecap="round"
                        stroke-linejoin="round"
                        stroke-width="1.5"
                      />
                    </svg>
                  </span>
                </button>
              </div>
            </div>
          </div>
          <button
            class="chakra-button css-1mv50kv"
            type="button"
          >
            Proceed
            <span
              class="css-9n3bbc"
            >
              <svg
                class="chakra-icon css-iz364c"
                focusable="false"
                viewBox="0 0 15 14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.25 7L13.75 7M13.75 7L8.125 1.375M13.75 7L8.125 12.625"
                  stroke="#33EFEF"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </span>
          </button>
        </div>
      </div>
      <p
        class="chakra-text css-1n4wwj3"
      />
    </div>
  </div>
</div>
`;
